website/src/components/SearchPage/DownloadDialog/DownloadUrlGenerator.ts

```diff
@@ -0,0 +1,114 @@
+    }
+
+    public generateDownloadUrl(downloadParameters: DownloadParameters, option: DownloadOption) {
+        const baseUrl = `${this.lapisUrl}${getEndpoint(option.dataType)}`;
+        const params = new URLSearchParams();
+
+        params.set('downloadAsFile', 'true');
+        params.set('downloadFileBasename', this.generateFilename(option.dataType));
+        if (!option.includeOldData) {
+            params.set(VERSION_STATUS_FIELD, versionStatuses.latestVersion);
+            params.set(IS_REVOCATION_FIELD, 'false');
+        }
+        if (!option.includeRestricted) {
+            params.set('dataUseTerms', 'OPEN');
+        }
+        if (option.dataType.type === 'metadata') {
+            params.set('dataFormat', metadataDefaultDownloadDataFormat);
+        }
+        if (option.compression !== undefined) {
+            params.set('compression', option.compression);
+        }
+
+        switch (downloadParameters.type) {
+            case 'filter':
+                const lapisSearchParameters = downloadParameters.lapisSearchParameters;
+                if (lapisSearchParameters.accession !== undefined) {
+                    for (const accession of lapisSearchParameters.accession) {
+                        params.append('accession', accession);
+                    }
+                }
+
+                const mutationKeys = [
+                    'nucleotideMutations',
+                    'aminoAcidMutations',
+                    'nucleotideInsertions',
+                    'aminoAcidInsertions',
+                ];
+
+                for (const [key, value] of Object.entries(lapisSearchParameters)) {
+                    // Skip accession and mutations
+                    if (key === 'accession' || mutationKeys.includes(key)) {
+                        continue;
+                    }
+                    const stringValue = String(value);
+                    const trimmedValue = stringValue.trim();
+                    if (trimmedValue.length > 0) {
+                        params.set(key, trimmedValue);
+                    }
+                }
+
+                mutationKeys.forEach((key) => {
+                    if (lapisSearchParameters[key] !== undefined) {
+                        params.set(key, lapisSearchParameters[key].join(','));
+                    }
+                });
+                break;
+            case 'select':
+                const sortedIds = Array.from(downloadParameters.selectedSequences).sort();
+                sortedIds.forEach((accessionVersion) => {
+                    params.append('accessionVersion', accessionVersion);
+                });
+                break;
+        }
+
+        return {
+            url: `${baseUrl}?${params}`,
+            baseUrl,
+            params,
+        };
+    }
+
+    private generateFilename(downloadDataType: DownloadDataType): string {
+        const siteName = kebabCase(this.websiteName);
+        const organism = kebabCase(this.organism);
```

Comment:
This will already be a "technical name" (if I understand correctly) - I'm not sure we need to do anything to it

Reply:
Might be nice to kebab just in case, but doesn't matter right now

Reply:
Yeah I agree so I'd keep it